src/planscape/forsys/generate_projects_for_a_single_scenario.R

```diff
@@ -1,88 +1,104 @@
 })
 
-generate_projects_for_a_single_scenario <- function(forsys_input_data,
-                                                    priorities,
-                                                    priority_weights,
-                                                    stand_id_field,
-                                                    proj_id_field,
-                                                    stand_area_field,
-                                                    stand_cost_field,
-                                                    geo_wkt_field,
-                                                    output_scenario_name,
-                                                    output_scenario_tag) {
-  wp_str <- "weighted_priorities"
-  # Enables debug mode if output_scenario_name and output_scenario_tag are
-  # non-empty.
+generate_projects_for_a_single_scenario <- function(
+  forsys_input_data,
+  priorities,
+  priority_weights,
+  stand_id_field,
+  proj_id_field,
+  stand_area_field,
+  stand_cost_field,
+  geo_wkt_field,
+  output_scenario_name,
+  output_scenario_tag,
+  PreForsysClusterType = 'KMEANS') {
+
+  # Enables debug mode if providing output_scenario_name and output_scenario_tag
   # If enabled, data and graphs are output to directory,
   # output/<output_scenario_name>/<output_scenario_tag>/
-  enable_debug <- nchar(output_scenario_name) > 0 && 
-    nchar(output_scenario_tag) > 0
-
-  priority_impact_fields <- paste(priorities, "_PCP", sep="")
+  enable_debug <- (
+    nchar(output_scenario_name) > 0 &&
+    nchar(output_scenario_tag) > 0)
 
   # Appends spm and pcp data columns for each priority.
   forsys_input_data <- forsys_input_data %>% 
-                forsys::calculate_spm(fields = priorities) %>%
-                forsys::calculate_pcp(fields = priorities)
+    forsys::calculate_spm(fields = priorities) %>%
+    forsys::calculate_pcp(fields = priorities)
+
   # Appends a preset_priority column, which contains total impact score values.
+  priority_impact_fields <- paste0(priorities, "_PCP")
+  wp_colname <- "weighted_priorities"
   suppressMessages(
     forsys_input_data <- forsys_input_data %>%
-                  forsys::combine_priorities(
-                    fields = priority_impact_fields, 
-                    weights = priority_weights, 
-                    new_field = wp_str)
-  )
+      forsys::combine_priorities(
+        fields = priority_impact_fields,
+        weights = priority_weights,
+        new_field = wp_colname))
+
   # Parses wkt in the geo_wkt column and adds it to a "geometry" column.
-  # Patchmax expects column name to be "geometry" - do not change the variable
-  # name.
-  # Internally, Patchmax calls st_sf on geometry column values. This results in
-  # an error unless st_as_sfc is called on the wkt first.
-  geometry <- lapply(forsys_input_data[geo_wkt_field], st_as_sfc)
-  forsys_input_data <- cbind(forsys_input_data, geometry)
+  forsys_input_data_formatted <- forsys_input_data %>%
```

Comment:
What's the difference between forsys_input_data and forsys_input_data_formatted?
Could they all be named forsys_input_data?
Alternatively, could forsys_input_data_formatted be the name of the variable assigned at line 31, after the calculate_spm and calculate_pcp calls?